core/src/main/java/org/infinispan/interceptors/impl/VersionedEntryWrappingInterceptor.java

```diff
@@ -29,33 +32,21 @@
    protected VersionGenerator versionGenerator;
    private static final Log log = LogFactory.getLog(VersionedEntryWrappingInterceptor.class);
 
-   @Override
-   protected Log getLog() {
-      return log;
-   }
-
-   @Inject
-   public void initialize(VersionGenerator versionGenerator) {
-      this.versionGenerator = versionGenerator;
-   }
+   private final InvocationSuccessFunction prepareHandler = this::prepareHandler;
 
-   @Override
-   public Object visitPrepareCommand(TxInvocationContext ctx, PrepareCommand command) throws Throwable {
-      VersionedPrepareCommand versionedPrepareCommand = (VersionedPrepareCommand) command;
-      if (ctx.isOriginLocal()) {
-         versionedPrepareCommand.setVersionsSeen(ctx.getCacheTransaction().getVersionsRead());
-      }
-      wrapEntriesForPrepare(ctx, command);
+   private Object prepareHandler(InvocationContext nonTxCtx, VisitableCommand command, Object nil) {
```

Comment:
I guess forgot to mention why I like having a method for every callback: because it allows us to keep related code together, and to minimize the diff when we extract part of a method into a callback :)